scripts/finetune_pfas.py

```
@@ -0,0 +1,172 @@
+    # Split dataset into training and validation sets
+    train_size = int(0.8 * len(full_dataset))
+    val_size = len(full_dataset) - train_size
+    train_dataset, val_dataset = torch.utils.data.random_split(full_dataset, [train_size, val_size])
+
+    train_loader = GraphDataLoader(train_dataset, batch_size=args.batch_size, shuffle=True)
+    val_loader = GraphDataLoader(val_dataset, batch_size=args.batch_size, shuffle=False)
+
+    # --- Fine-tuning Setup ---
+    optimizer = optim.Adam(model.parameters(), lr=args.lr)
+    loss_fn = torch.nn.L1Loss()  # Use MAE loss
+    scheduler = ReduceLROnPlateau(optimizer, mode='min', factor=0.8, patience=5, verbose=True)
```

Ruff (0.12.2)

Undefined name `ReduceLROnPlateau`

(F821)